pkg/pool-hooks/contracts/hooks-quantamm/HyperSurgeHook.sol

```diff
@@ -0,0 +1,747 @@
+    uint256 private immutable _defaultCapDeviation;
+
+    constructor(
+        IVault vault,
+        uint256 defaultMaxSurgeFeePercentage,
+        uint256 defaultThresholdPercentage,
+        uint256 defaultCapDeviation,
+        string memory version
+    ) SingletonAuthentication(vault) VaultGuard(vault) Version(version) {
+        _ensureValidPct(defaultMaxSurgeFeePercentage);
+        _ensureValidPct(defaultThresholdPercentage);
+        _ensureValidPct(defaultCapDeviation);
+        _defaultMaxSurgeFee = defaultMaxSurgeFeePercentage;
+        _defaultThreshold = defaultThresholdPercentage;
+        _defaultCapDeviation = defaultCapDeviation;
+    }
+
+    ///@inheritdoc IHooks
+    function getHookFlags() external pure override returns (HookFlags memory hookFlags) {
+        hookFlags.shouldCallComputeDynamicSwapFee = true;
+        hookFlags.shouldCallAfterAddLiquidity = true;
+        hookFlags.shouldCallAfterRemoveLiquidity = true;
+    }
+
+    /// @inheritdoc IHooks
+    function onRegister(
+        address,
+        address pool,
+        TokenConfig[] memory tokenCfgs,
+        LiquidityManagement calldata
+    ) external override onlyVault returns (bool) {
+        PoolDetails memory details;
+        if (tokenCfgs.length >= 2 && tokenCfgs.length <= 8) {
+            details.arbMaxSurgeFeePercentage = _defaultMaxSurgeFee.toUint32();
+            details.arbThresholdPercentage = _defaultThreshold.toUint32();
+            details.arbCapDeviationPercentage = _defaultCapDeviation.toUint32();
+            details.noiseMaxSurgeFeePercentage = _defaultMaxSurgeFee.toUint32();
+            details.noiseThresholdPercentage = _defaultThreshold.toUint32();
+            details.noiseCapDeviationPercentage = _defaultCapDeviation.toUint32();
+
+            details.numTokens = uint8(tokenCfgs.length);
+
+            details.initialized = true;
+            _poolCfg[pool].details = details;
+
+        } else {
+            revert NumTokensOutOfRange();
+        }
+
+        return true;
+    }
+
+    /// @notice Configure a single token’s Hyperliquid mapping for a given pool by token index (0..7).
+    /// @param pool The pool address to configure.
+    /// @param tokenIndex The index of the token to configure (0..7).
+    /// @param pairIdx the index of the pair being set
+    function setTokenPriceConfigIndex(
+        address pool,
+        uint8 tokenIndex,
+        uint32 pairIdx
+    ) external onlySwapFeeManagerOrGovernance(pool) {
+        //TODO should this be done on construction? Not sure there is any reason to change it
+        //or at least be blocked once set
+        TokenPriceCfg memory tempCfg;
+        PoolDetails memory details = _poolCfg[pool].details;
+
+        if (!details.initialized) {
+            revert PoolNotInitialized();
+        }
+        if (tokenIndex >= details.numTokens) {
+            revert TokenIndexOutOfRange();
+        }
+
+        if (pairIdx == 0) {
+            revert InvalidPairIndex();
+        }
```

Comment:
This could be at the beginning of the function, would avoid a storage read of `_poolCfg[pool].details`.